syntax = "proto3";

package admin_course;

service AdminCourseService {
    rpc AdminCreateCourse (AdminCreateCourseRequest) returns (AdminCreateCourseResponse);
    rpc AdminGetCourse    (AdminGetCourseRequest)    returns (AdminGetCourseResponse);
    rpc AdminUpdateCourse (AdminUpdateCourseRequest) returns (AdminUpdateCourseResponse);
    rpc AdminDeleteCourse (AdminDeleteCourseRequest) returns (AdminDeleteCourseResponse);
    rpc AdminListCourses  (AdminListCoursesRequest)  returns (AdminListCoursesResponse);
}

message AdminCreateCourseRequest {
    string title = 1;
    string description = 2;
    string preview_url = 3;
    string mentor_id = 4;
    string category_id = 5;
}

message AdminCreateCourseResponse {
    string id = 1;
}

message AdminGetCourseRequest {
    string id = 1;
}

message AdminGetCourseResponse {
    string id = 1;
    string title = 2;
    string description = 3;
    string preview_url = 4;
    string mentor_id = 5;
    string category_id = 6;
}

message AdminUpdateCourseRequest {
    string id = 1;
    optional string title = 2;
    optional string description = 3;
    optional string preview_url = 4;
    optional string mentor_id = 5;
    optional string category_id = 6;
}

message AdminUpdateCourseResponse {
    string id = 1;
}

message AdminDeleteCourseRequest {
    string id = 1;
}

message AdminDeleteCourseResponse {
    bool success = 1;
}


message AdminListCoursesRequest {
    optional string category_id = 1;
}

message AdminListCoursesResponse {
    repeated AdminGetCourseResponse items = 1;
}
